build.sbt

```diff
@@ -7,7 +7,7 @@ version := "0.0.1"
 scalaVersion := "2.12.15"
  
 libraryDependencies ++= Seq(
-  "org.apache.daffodil" %% "daffodil-tdml-processor" % "3.2.0" % "test",
+  "org.apache.daffodil" %% "daffodil-tdml-processor" % "3.5.0-SNAPSHOT" % "test",
```

Comment:
Normally we would do updates like this on a dev branch. But this constant churn of the schema projects has proven so problematic that I'm thinking the version should be 'currentSnapshot' (or whatever that magic token is, until we tag something, at that point it should be wired to a specific version. 

I'm ok just changing this to 3.5.0-SNAPSHOT for now.

Reply:
This change to use Runner instead of DFDLTestSuite is old and should be fine with 3.4.0. I'd prefer we use that released version so this works for people not developing daffodil. And then we can figure out the best way to make this easily overridable and testable with snapshots later.

Reply:
My goal here is to update the import in the test scala file since that changed with the OSGI refactoring for 3.5. This is only the first schema I checked so there will be many more that will probably need a similar change. What is the best way to update this repository in anticipation of the 3.5 changes? I don't see a dev branch here. 

As for the DFDLTestSuite vs Runner, I got this error when moving the version up to 3.5.0-SNAPSHOT:

```
TestMarbles.scala:10:21: constructor DFDLTestSuite in class DFDLTestSuite cannot be accessed in object TestMarbles
```

That error is also present in 3.4 but absent when I roll back to 3.2. I'm trying to see what the difference is that is causing the error but I'm missing something. I made the change to the Runner because of what the comment said and since I couldn't get it to compile.

Reply:
> My goal here is to update the import in the test scala file since that changed with the OSGI refactoring for 3.5. 

The only import that changed due to OSGI was `Misc`, and that is no longer needed with the update to use `Runner`. So I think this file shouldn't be affected by OSGI changes, and if it works on 3.4.0 then it should work on 3.5.0. 

> As for the DFDLTestSuite vs Runner, I got this error 

I think DFDLTestSuite was was deprecated and access removed a while ago, though I'm not sure which version. Updating to `Runner` is the correct thing to do, and should work on both 3.4.0 and 3.5.0. Note that `validateTDMLFile`, `validateDFDLSchemas`, default to `true`, so only `compileAllTopLevel` is needed (if we want it, I'm not sure how important that is for schemas where we know all global decls are used).

Reply:
Another thought, I think there is value in updated everything to 3.4.0 first so we get an idea of what is actually broken in 3.5.0. I don't think the "public" parts of Daffodil (e.g. SAPI, JAPI, TDML Runner) changed due to OSGI, so as long as schema projects only rely on only those, then maybe a 4.0.0 isn't needed, and many of the breakages we are seeing are old.

Reply:
> The only import that changed due to OSGI was `Misc`, and that is no longer needed with the update to use `Runner`. So I think this file shouldn't be affected by OSGI changes, and if it works on 3.4.0 then it should work on 3.5.0.

Thanks, I totally missed that the second change removed the need to have the first change. It does work on 3.4.0 when it is updated to the Runner. I've changed the version number in build.sbt to that. 

> Another thought, I think there is value in updated everything to 3.4.0 first so we get an idea of what is actually broken in 3.5.0

That's a good idea.